client/src/components/Auth/AuthForm.tsx

```diff
@@ -1,121 +1,163 @@
-import { useState, useRef, } from 'react';
-import { useAppDispatch } from '../../store/hook';
-import { useNavigate } from 'react-router-dom';
-import Axios,{AxiosError} from 'axios';
-import { authActions } from '../../store/auth';
-import classes from './AuthForm.module.css';
-import { ToastContainer, toast } from 'react-toastify';
-import 'react-toastify/dist/ReactToastify.css';
+import { useEffect, useState } from "react";
+import { useNavigate } from "react-router-dom";
+import { useAppDispatch } from "../../store/hook";
+import useInput, { IUserInput } from "../../hooks/use-input";
+import useHttp from "../../hooks/use-http";
+import { authActions } from "../../store/auth";
+import { toastActions } from "../../store/toast";
+import { Title } from "../../store/toast";
+import Input from "../UI/Input";
+import Button from "../UI/Button";
+import { loginURL, registerURL } from "../../config";
+import classes from "./AuthForm.module.css";
 
 const AuthForm = () => {
   const navigate = useNavigate();
   const dispatch = useAppDispatch();
-  const nameInputRef = useRef<HTMLInputElement>(null);
-  const emailInputRef = useRef<HTMLInputElement>(null);
-  const passwordInputRef = useRef<HTMLInputElement>(null);
+  const { isLoading, errorMessage, errCount, data, sendRequest: signIn } = useHttp();
+  const {
+    value: name,
+    isValid: nameIsValid,
+    hasError: nameHasError,
+    valueChangeHandler: nameChangeHandler,
+    valueBlurHandler: nameBlurHandler,
+    reset: nameResetHandler,
+  }: IUserInput = useInput((value: string) => value.trim() !== "");
 
+  const {
+    value: email,
+    isValid: emailIsValid,
+    hasError: emailHasError,
+    valueChangeHandler: emailChangeHandler,
+    valueBlurHandler: emailBlurHandler,
+    reset: emailResetHandler,
+  }: IUserInput = useInput((value: string) => value.includes("@"));
+
+  const {
+    value: password,
+    isValid: passwordIsValid,
+    hasError: passwordHasError,
+    valueChangeHandler: passwordChangeHandler,
+    valueBlurHandler: passwordBlurHandler,
+    reset: passwordResetHandler,
+  }: IUserInput = useInput((value: string) => value.trim() !== "");
 
   const [isLogin, setIsLogin] = useState(true);
-  const [isLoading, setIsLoading] = useState(false);
 
   const switchAuthModeHandler = () => {
     setIsLogin((prevState) => !prevState);
   };
 
+  let formIsValid = false;
+  if (isLogin) {
+    if (emailIsValid && passwordIsValid) formIsValid = true;
+  } else {
+    if (nameIsValid && emailIsValid && passwordIsValid) formIsValid = true;
+  }
+
```

Comment:
i would suggest exploring yup and formik npm packages later for forms and validations to make this code a lot more cleaner